### ISSUE TITLE
access_denied after update from Kentor.AuthServices.Owin 0.17.2 to Sustainsys.Saml2.Owin 0.24.0

### ISSUE THREAD

**Issue by parkinsona** [Sustainsys/Saml2#952]
Hi,

I am updating my current identityserver 3 project to use updated nuget packages. I have done this for both identity server and Saml2 (previously Kentor.AuthServices).

Prior to the change my SAML2 okta confiugrations were working. 
However, after the update I am getting access_denied when Okta is posting back https://localhost:44333/core/OktaLocal2/acs (which I believe is my single sign on url). The next request is then: https://localhost:44333/core/callback?error=access_denied
As far as I can tell the saml2 response is as it should be.

If I attempt to go to https://localhost:44333/core/OktaLocal2/acs it gives me access_denied. Where previously it would give an error saying that it couldn't load without a saml response. 

Has something changed that would cause this issue to happen?

Here are some things I tried...
- cookie saver - no difference
- changing encryption to RA-SHA256, it let me login, however when I got to my relying party, it never created the cookie correctly.

**Comment by parkinsona** [comment]
@AndersAbel , do you have any ideas on this?

**Comment by AndersAbel** [comment]
Enable logging, a detailed error message should be visible in the logs.

**Comment by Peperud** [comment]
@parkinsona 
Same situation as you (IdSrv3 and upgrading packages) and this bit me yesterday. I was just about to enter a new issue for it. 
Change the 'acs' to '**A**cs'. 

@AndersAbel 
The error I was getting in the logs was very misleading:
`No Idp with entity id "https://pingone.com/idp/my-test found.
The given key was not present in the dictionary.
`

**Comment by parkinsona** [comment]
Thanks @AndersAbel,  logging revealed the following error: 

<code>

Sustainsys.Saml2.Exceptions.InvalidSignatureException: The signing algorithm http://www.w3.org/2000/09/xmldsig#rsa-sha1 is weaker than the minimum accepted http://www.w3.org/2001/04/xmldsig-more#rsa-sha256. If you want to allow this signing algorithm, use the minIncomingSigningAlgorithm configuration attribute.

</code>

I tried again to change the encryption on the Okta provider and it worked this  time round. Not sure why it liked it this time. I also played with the minIncomingSigningAlgorithm setting and that also helped. 

Now I just need to work out if I set the minimum for my providers or see if they can change.
